pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -181,6 +180,27 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, nil
 	}
 
+	// Get the private key that will be used to configure the instance
```

Comment:
Should this code block be moved before fetching machine instance? We should be checking for private key before checking machine status. It would allow us to see why it did not go in `Provisioned` state.

Reply:
agree, I think moving it up makes sense

Reply:
This has not been addressed.

Reply:
You're right, i had moved it up but not enough, doing that now